src/classes/TriggerHandler.cls

```diff
@@ -0,0 +1,102 @@
+/**
+* Description   - Created a handler class To add prefix and to send mail
+* Created by    - Sudheer Kumar Shakya
+* Created Date  - 27/07/2022
+* Revision log  -  V_1.0 - Created  - Sudheer Kumar Shakya - 27/07/2022
+*                  V_1.1 - Modified - Sudheer Kumar Shakya - 27/07/2022 
+**/
+public with sharing class TriggerHandler {
+
+    /**   Description :   Method for adding and updating Prefix According to its Type or changes 
+    *     Created By  :   Sudheer Kumar Shakya
+    *     Arguments   :   List(trigger.new)
+    *     Return      :   void
+    *     Revision log:   V_1.0 - Created  - Sudheer Kumar Shakya - 27/07/2022
+    *                     V_1.1 - Modified - Sudheer Kumar Shakya - 27/07/2022 
+    **/
+    public static void prefixforAccount(List<Account> newAccount, Map<Id,Account> oldAccountMap){
+       
+        for(Account account: newAccount){
+           
+            //Generating a Random Account Number
+            if(account.AccountNumber==null){
+                Integer accnum=Integer.valueOf(100000000*Math.random());
+                account.AccountNumber=String.valueOf(accnum);     
+            }
+
+            //Adding Prefix to account Number accordingly:
+            if(oldAccountMap == null || account.Type != oldAccountMap.get(account.Id).Type ){
+                if(account.Type=='Saving Account'){
+                    if(account.AccountNumber.contains('FD-')){
+                        account.AccountNumber=account.AccountNumber.remove('FD-') ;   
+                    }  
+                    account.AccountNumber='SA-'+account.AccountNumber;
+                }
+
+                else if(account.Type=='Fixed Deposit'){
+                    if(account.AccountNumber.contains('SA-')){
+                        account.AccountNumber=account.AccountNumber.remove('SA-') ;   
+                    }  
+                    account.AccountNumber='FD-'+account.AccountNumber;
+                }    
+            
+                else if(account.Type!='Fixed Deposit' && account.Type!='Saving Account'){
```

Comment:
Just use else condition here. If it is either fix Deposit or Saving then it will never come in else condition as either above if condition meets or else if condition.